sky/jobs/server/core.py

```diff
@@ -256,7 +261,20 @@ def _maybe_restart_controller(
     rich_utils.force_update_status(
         ux_utils.spinner_message(f'{spinner_message} - restarting '
                                  'controller'))
-    handle = sky.start(jobs_controller_type.value.cluster_name)
+    handle = core.start(cluster_name=jobs_controller_type.value.cluster_name)
+    # Make sure the dashboard is running when the controller is restarted.
+    # We should not directly use execution.launch() and have the dashboard cmd
+    # in the task setup because since we are using detached_setup, it will
+    # become a job on controller which messes up the job IDs (we assume the
+    # job ID in controller's job queue is consistent with managed job IDs).
+    logger.info('Starting dashboard...')
+    runner = handle.get_command_runners()[0]
+    runner.run(
+        f'export '
+        f'{skylet_constants.USER_ID_ENV_VAR}={common_utils.get_user_hash()!r}; '
+        f'{managed_job_constants.DASHBOARD_SETUP_CMD}',
+        stream_logs=True,
+    )
```

Comment:
after #4538 we only need to do this for k8s, I think

Reply:
It might be fine to run this, as we have the checking in the setup command for whether a service or python should be used.

Reply:
Ah I didn't see that. Yes, seems ok.